### ISSUE TITLE
Please move crash.log somewhere else

### ISSUE THREAD

**Issue by engenegr** [ElementsProject/lightning#4458]
### Issue and Steps to Reproduce

The issue is simple. I've removed the wallet database along with crash.log.* when did `rm * crash.log.202*` and missed extra `*`.

**Comment by rustyrussell** [comment]
Ouch!

I don't think there's a good other place to put them though?

I think we're best off moving forward with something like @llfourn's deterministic basepoints for something like this.  That way we could claw back channel discovery even if you only have a seed backed up somewhere.

**Comment by engenegr** [comment]
You already have `debug.log` in `.lightning` and its location may be changed via configuration. So I was always amused why there are other logs somewhere else.

**Comment by jsarenik** [comment]
Good point @engenegr. The log output dir could be configurable and even `crash.log`s would fall in there.

Currently there is only `--log-file=` option and maybe something like `--log-dir=` would make sense. What do others think?

**Comment by vincenzopalazzo** [comment]
Another solution could be put the crash log in the directory of the log file, e.g: /home/user/log.log -> and put the crash log in /home/user/.

**Comment by jsarenik** [comment]
There is actually a neat thing in `lightningd/log.c` already which does not require any change.

Make the directory (e.g. `~/.lightning/testnet`) read-only and the `crash.log` will be written to `/tmp`.

**Comment by jsarenik** [comment]
@engenegr What do you think about above? [^^](https://github.com/ElementsProject/lightning/issues/4458#issuecomment-810396853)

**Comment by engenegr** [comment]
@jsarenik likely the only way for the user to know this is to remove wrong files once.

Also, I do not think making `testnet` and `bitcoin` read-only is the solution. There are also sqlite files and they are changing.

**Comment by jsarenik** [comment]
> @jsarenik likely the only way for the user to know this is to remove wrong files once.
> 
> Also, I do not think making `testnet` and `bitcoin` read-only is the solution. There are also sqlite files and they are changing.

@engenegr Yes, it is not documented I suppose.

Making the directory read-only only makes new files in that directory impossible.

Try this:

```
cd /tmp
mkdir mytest$$
cd mytest$$
mkdir rodir
echo test > rodir/datafile
chmod a-w rodir
ls -ld rodir
echo additional >> rodir/datafile
echo test > rodir/newfile
```

After the last command you should get `Permission denied`

As the owner you can always change the file/directory mode bits (`man chmod`).

**Comment by sangbida** [comment]
This issue is being closed as part of issue cleanup.

**Reason**: No recent activity (last updated 1777 da

If you believe this issue should remain open, please comment and we can reopen it.
